plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/SQLGroupingQueryGenerator.java

```diff
@@ -205,4 +215,57 @@ public String[] getFuncAliases() {
         return funcAliases;
     }
 
+    /**
+     * Represents a formatted table in an SQL query.
+     * This class is used to format the table name according to the specified SQL dialect.
+     * It is necessary because `net.sf.jsqlparser.schema.Table` only supports a dot as a separator.
+     */
+    private record FormattedTable(Table table, SQLDialect sqlDialect) implements FromItem {
+
+        @Override
+        public void accept(FromItemVisitor fromItemVisitor) {
+            table.accept(fromItemVisitor);
+        }
+
+        @Override
+        public Alias getAlias() {
+            return table.getAlias();
+        }
+
+        @Override
+        public void setAlias(Alias alias) {
+            table.setAlias(alias);
+        }
+
+        @Override
+        public Pivot getPivot() {
+            return table.getPivot();
+        }
+
+        @Override
+        public void setPivot(Pivot pivot) {
+            table.setPivot(pivot);
+        }
+
+        @Override
+        public UnPivot getUnPivot() {
+            return table.getUnPivot();
+        }
+
+        @Override
+        public void setUnPivot(UnPivot unPivot) {
+            table.setUnPivot(unPivot);
+        }
+
+        @Override
+        public String toString() {
+            String tableName = table.getDatabase()
```

Comment:
Both database and schema are optional.
Also maybe it is better to extend net.sf.jsqlparser.schema.Table than implement interface (besides other issues interface changes in every major version we'll have to reimplement it)